app/src/main/java/io/neurolab/fragments/FocusVisualFragment.java

```diff
@@ -324,6 +326,13 @@ private void buildInstructionDialog() {
     @Override
     public void onResume() {
         super.onResume();
+        IntentFilter intentFilter = new IntentFilter();
+        // adding the possible USB intent actions.
+        intentFilter.addAction(ACTION_USB_PERMISSION);
+        if (dataReceiver != null && !isReceiverRegistered) {
+            getContext().registerReceiver(dataReceiver, intentFilter);
```

Comment:
The receiver will be again registered on resume and data transmission will start all over again with new intent filter and perms. Let's not do any of these changes right now.